### PR TITLE
Serial port Modbus-RTU protocol implementation

### PR DESCRIPTION
**Pull request by yureng**
I added SerialPort Modbus-RTU to the library section of the nf-Community-Contributions

## Checklist:

- [x] I've placed my contribution in the appropriate section.
- [x] I've choosen a good name for it and I've checked the repo for duplicates.
- [x] I've added a readme.md following the provided template (TBD). 
- [x] My code follows the code style of this project.
- [x] I've tidy my files inside the contribution folder.
- [x] I've edited the repo README.md to include the mention and link to my contribution. It's under the proper section and the link is correct.
- [x] The code builds.

### CODE REVIEW

File: libraries/nanoFramework.Modbus/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# nanoFramework.Modbus
```
Comment (Ellerbach) [inline libraries/nanoFramework.Modbus/README.md:1]: Markdown 101: you need a new line after a title
```suggestion
# nanoFramework.Modbus

```

File: libraries/nanoFramework.Modbus/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+Serial port Modbus-RTU protocol implementation.
+
+Repository: [https://github.com/yureng/nanoFramework.Modbus](https://github.com/yureng/nanoFramework.Modbus).<br>
```
Comment (Ellerbach) [inline libraries/nanoFramework.Modbus/README.md]: Markdown 101: if you need a new line, make the new line, don't use HTML code for that
```suggestion
Repository: [https://github.com/yureng/nanoFramework.Modbus](https://github.com/yureng/nanoFramework.Modbus).

```

File: libraries/nanoFramework.Modbus/README.md [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+Sample is [here](https://github.com/yureng/nanoFramework.Modbus/blob/main/src/ESP32_Demo/Program.cs).<br>
```
Comment (Ellerbach) [inline libraries/nanoFramework.Modbus/README.md]: Markdown 101: you need a new empty line at the end of the file.
```suggestion
Sample is [here](https://github.com/yureng/nanoFramework.Modbus/blob/main/src/ESP32_Demo/Program.cs).

```

**Review by Ellerbach — COMMENTED** [review]
Looks good to me, few Markdown suggestions :-) Question: would you be interested in adding your Modbus implementing into the https://github.com/nanoframework/nanoFramework.IoT.Device repository? It will then be available as a nuget for everyone. Code style wise, you will have to adjust few elements and add few more intellisense (the /// ) comments on some of the public elements, also translate some of the public intellisense into English. I'll be happy to help and assist in the process.

File: libraries/nanoFramework.Modbus/README.md (hunk 1 above)
Comment (yureng) [reply libraries/nanoFramework.Modbus/README.md:1]: modified, please review

Comment (yureng) [reply libraries/nanoFramework.Modbus/README.md:1]: I would be honored to be able to add Modbus implementing to the nanoFramework.IoT.Device repository.

I need some time to standardize and modify the code style, and when everything is ready, I pull request

Comment (Ellerbach) [reply libraries/nanoFramework.Modbus/README.md:1]: Whenever you'll be ready, go for a PR. Happy to help you and assist you.

**Review by Ellerbach — APPROVED** [review]

**Comment by josesimoes** [comment]
@yureng I was wondering if your code has settled by now. 😁 
Let's have this added as an official library for nanoFramework.

Disclaimer: I have particular insterest in this as I'm working on a project that requires modbus 😉 
Therefore I can assign time to help on this. 

Don't worry about the code not being perfect or some tiny bit missing. Just go for a PR in IoT bindings repo and we'll guide you through the process.

**Comment by josesimoes** [comment]
@yureng any feedback on this?

**Comment by yureng** [comment]
@josesimoes  Sorry for the delay.

I've submitted a PR:  [https://github.com/nanoframework/nanoFramework.IoT.Device/pull/838]
but StyleCop is confusing me so much that I don't know how to correct it

Because my English is not good and I need to rely on machine translation, I may have a wrong understanding of a large number of StyleCop prompts

**Comment by josesimoes** [comment]
@yureng got it! Thanks for taking the time. 👍🏻 
Let's continue there. I'm sure we can help you fixing things with stylecop.

**Pull request merged**